src/renderer/entities/staking/lib/useValidatorsMap.ts

```diff
@@ -0,0 +1,37 @@
+import { useEffect, useState } from 'react';
+import { ApiPromise } from '@polkadot/api';
+
+import { ValidatorMap, useEra, useValidators } from './';
+import { ChainId } from '@renderer/shared/core';
+
+export const useValidatorsMap = (api?: ApiPromise, chainId?: ChainId, isLightClient?: boolean): ValidatorMap => {
+  const { subscribeActiveEra } = useEra();
+  const { getValidatorsWithInfo } = useValidators();
+
+  const [era, setEra] = useState<number>();
+  const [validators, setValidators] = useState<ValidatorMap>({});
+
+  useEffect(() => {
+    let unsubEra: () => void | undefined;
+
+    if (api) {
+      (async () => {
+        unsubEra = await subscribeActiveEra(api, setEra);
```

Comment:
`subscribeActiveEra(api, setEra).then(unsubFn => unsubEra = unsubFn);`
Would be a bit cleaner than `iife async await`